### PR TITLE
Add fromMnemonic function

### LINKED ISSUE #178: different ETH keys for the same mnemonic
Hi everyone.
I use golang library for creating HD wallets: 

```golang
import (
	"fmt"
	"log"

	"github.com/lizc2003/hdwallet/wallet"
)

mnemonic := "thumb...."
w, err = hdw.NewWallet(wallet.SymbolEth, 0, 0, 0)
	if err != nil {
		log.Fatal(err)
	}
	fmt.Printf("wallet: %s\n\tprivatekey: %s\n\tpublickey: %s\n\taddress: %s\n",
		w.Symbol(),
		w.DerivePrivateKey(),
		w.DerivePublicKey(),
		w.DeriveAddress())	

```

as derived address I got: `address: 0x05D29c8C..................8DAb2ff43e2e3C310`

I want to move my solution to ethereumjs, and I use the same mnemonic

```js
const {hdkey} = require("ethereumjs-wallet")

let mnemonic = "thumb...."

let buffer = Buffer.from(mnemonic)
let hd_priv_key = hdkey.fromMasterSeed(buffer)
let address = hd_priv_key.deriveChild(0).getWallet().getAddressString()

console.log(address)
```

and I got `0xd5b8f015............8a481933ba2a8336`

what is the difference, 

note: the same address as for golang I got on https://iancoleman.io/bip39/

could you explain and help with it?

### PR DESCRIPTION
**Pull request by kaliubuntu0206**
closes #178

Requiring BIP39 dependency is useless since the functionality of converting BIP39 mnemonic phrases are already in the imported package.

And, lower count of dependency => better security for the codebase

### CODE REVIEW

**Comment by kaliubuntu0206** [comment]
cc @acolytec3 @paulmillr @holgerd77

Thank you for your great works of the crypto libraries though @paulmillr you are doing a god's work

File: src/hdkey.ts [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
    */
   public static fromMasterSeed(seedBuffer: Buffer): EthereumHDKey {
     return new EthereumHDKey(HDKey.fromMasterSeed(seedBuffer))
   }
 
+  /**
+   * Creates an instance based on BIP39 mnemonic phrases
+   */
+  public static fromMnemonic(mnemonic: string, passphrase?: string) {
+    return new EthereumHDKey(HDKey.fromMasterSeed(mnemonicToSeedSync(mnemonic, passphrase)))
```
Comment (paulmillr) [inline src/hdkey.ts:17]: `EthereumHDKey.fromMasterSeed(mnemonicToSeedSync(mnemonic, passphrase))`

**Review by paulmillr — APPROVED** [review]

**Comment by paulmillr** [comment]
this is fine, however, the repository is deprecated and has been merged into ethereumjs-monorepo.

So, we need to archive this gh repo.

Would it be possible for you to submit a pull request to monorepo?

**Comment by kaliubuntu0206** [comment]
@paulmillr Done https://github.com/ethereumjs/ethereumjs-monorepo/pull/2992

**Comment by kaliubuntu0206** [comment]
Closing as it is not going to merged I guess, for anyone who wants this feature you can use the latest @ethereumjs/wallet package https://github.com/ethereumjs/ethereumjs-monorepo/releases/tag/%40ethereumjs%2Fwallet%402.0.1 or ethers.js

**Pull request closed without merging**